components/filter/filter-dimension-set.js

```diff
@@ -113,15 +119,62 @@ class FilterDimensionSet extends LitElement {
 		return values;
 	}
 
+	_dispatchEvent(eventName, eventDetail) {
+		/** @ignore */
+		this.dispatchEvent(new CustomEvent(eventName, {
+			detail: eventDetail,
+			bubbles: true,
+			composed: false
+		}));
+	}
+
+	_getEmptyState(emptyStateSlot) {
+		if (!emptyStateSlot) return null;
+		const nodes = emptyStateSlot.assignedNodes({ flatten: true });
+		const emptyState = nodes.find((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName.toLowerCase() === 'd2l-filter-dimension-set-empty-state');
+		if (!emptyState) return null;
+		return {
+			actionHref: emptyState.actionHref,
+			actionText: emptyState.actionText,
+			description: emptyState.description
+		};
+	}
+
+	_getSlottedNodes() {
+		if (!this._slot) return [];
+		const nodes = this._slot.assignedNodes({ flatten: true });
+		return nodes.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName.toLowerCase() === 'd2l-filter-dimension-set-value');
+	}
+
+	_handleDimensionSetSearchEmptyStateChange(e) {
+		e.stopPropagation();
+		this._dispatchEvent('d2l-filter-dimension-empty-state-change', { dimensionKey: this.key, type: EmptyStateType.Search, changes: e.detail.changes });
+	}
+
+	_handleDimensionSetSetEmptyStateChange(e) {
+		e.stopPropagation();
+		this._dispatchEvent('d2l-filter-dimension-empty-state-change', { dimensionKey: this.key, type: EmptyStateType.Set, changes: e.detail.changes });
+	}
+
 	_handleDimensionSetValueDataChange(e) {
 		e.stopPropagation();
-		this._dispatchDataChangeEvent({ dimensionKey: this.key, valueKey: e.detail.valueKey, changes: e.detail.changes });
+		this._dispatchEvent('d2l-filter-dimension-data-change', { dimensionKey: this.key, valueKey: e.detail.valueKey, changes: e.detail.changes });
+	}
+
+	_handleSearchEmptyStateSlotChange(e) {
+		if (!this._searchEmptyStateSlot) this._searchEmptyStateSlot = e.target;
+		this._dispatchEvent('d2l-filter-dimension-empty-state-slot-change', { dimensionKey: this.key, type: EmptyStateType.Search });
```

Comment:
The approach I took here creates a separate slot change event for the empty state that the d2l-filter handles differently from when the dimension-set-empty-state attributes change. This is slightly different from how the filter handles dimension-set-value changes where the dimension-set will dispatch the same event for both attribute changes and slot changes.

I originally tried to do it the same way with a single `d2l-filter-dimension-empty-state-change` event but I thought the resulting handler in `d2l-filter` was doing too much and it was cleaner to separate them into separate handlers.